src/Raven.Client/Documents/Subscriptions/SubscriptionCriteria.cs

```diff
@@ -25,6 +25,73 @@ public SubscriptionCriteria(string collection)
 
     public class SubscriptionCriteria<T>
     {
+        private class LinqMethodsSupport : JavascriptConversionExtension
+        {
+            public override void ConvertToJavascript(JavascriptConversionContext context)
+            {
+                var methodCallExpression = context.Node as MethodCallExpression;
+                var methodName = methodCallExpression?
+                    .Method.Name;
+
+                if (methodName == null)
+                    return;
+
+                string newName;
+                switch (methodName)
+                {
+                    case "Any":
+                        newName = "some";
+                        break;
+                    case "All":
+                        newName = "every";
+                        break;
+                    case "Select":
+                        newName = "map";
+                        break;
+                    case "Where":
+                        newName = "filter";
+                        break;
+                    case "Contains":
```

Comment:
This is wrong, I think. `Contains` returns bool, but `indexOf` returns int. How does this work?